adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -1870,51 +1870,106 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] extends Logging {
+
+          optRegions.map(rp => {
+            val (firstRegion, secondRegion) = rp
+
+            // in the case where we span multiple referenceNames using
+            // IntervalArray.get with requireOverlap set to false will assign all
+            // the remaining regions to this partition, in addition to all the
+            // regions up to the start of the next partition.
+            if (firstRegion.referenceName != secondRegion.referenceName) {
+
+              // this partition covers the end of one contig and the
+              // start of another
+              Iterable((ReferenceRegion.toEnd(firstRegion.referenceName,
+                firstRegion.start), index),
+                (ReferenceRegion.fromStart(secondRegion.referenceName,
+                  secondRegion.end), index))
+            } else {
+              // otherwise we just have the ReferenceRegion span from partition
+              // lower bound to upper bound.
+              // We cannot use the firstRegion bounds here because we may end up
+              // dropping data if it doesn't map anywhere.
+              Iterable((ReferenceRegion(
+                firstRegion.referenceName,
+                firstRegion.start,
+                secondRegion.end),
+                index))
+            }
+          })
+        }).flatten
+          .groupBy(_._1.referenceName)
+          .map(p => {
+            val (referenceName, indexedRegions) = p
+
+            val regions: Seq[(ReferenceRegion, Int)] = if (indexedRegions.size == 1) {
+              // if we only have a single partition for this contig, extend the
+              // region to cover the whole contig
+              Seq((ReferenceRegion.all(indexedRegions.head._1.referenceName),
+                indexedRegions.head._2))
+            } else {
+              val sortedRegions = indexedRegions.sortBy(_._2)
+
+              // if we have multiple partitions for this contig, extend the
+              // first region to the start of the contig, and the last region to
+              // the end of the contig
+              sortedRegions.take(1).map(rp => {
+                (ReferenceRegion.fromStart(rp._1.referenceName,
+                  rp._1.end), rp._2)
+              }) ++ sortedRegions.tail.dropRight(1) ++ sortedRegions.takeRight(1)
+                .map(rp => {
+                  (ReferenceRegion.toEnd(rp._1.referenceName,
+                    rp._1.start), rp._2)
+                })
+            }
+
+            (referenceName, regions)
+          })
+
+      // the above map does not contain sequences who exist in the sequence
+      // dictionary but who are not seen in a record at the start/end of a
+      // partition
+      //
+      // here, we loop over the sequence records, check if they are in the map,
+      // and create a record if the sequence is not in the map
+      var lastIdx = 0
+      val missingSequences: Map[String, Seq[(ReferenceRegion, Int)]] =
+        sequences.records
```

Comment:
there wasn't a dependency on `sequences` before, will this still work if both RDDs have empty sequence dictionaries?  I can give it a try with two `FeatureRDD`s

Reply:
Yeah, we can't guarantee correct behavior if you don't have correctly defined sequence dictionaries.